evalml/tests/automl_tests/test_automl.py

```diff
@@ -1904,3 +1909,17 @@ class Pipeline2(TimeSeriesRegressionPipeline):
             continue
         assert result['parameters']['Delayed Feature Transformer'] == configuration
         assert result['parameters']['pipeline'] == configuration
+
+
+@patch('evalml.pipelines.TimeSeriesRegressionPipeline.score')
+@patch('evalml.pipelines.TimeSeriesRegressionPipeline.fit')
+def test_automl_best_pipeline_false(mock_fit, mock_score, X_y_binary):
+    X, y = X_y_binary
+
+    automl = AutoMLSearch(problem_type='binary', train_best_pipeline=False)
+    automl.search(X, y)
+    assert not automl.best_pipeline._is_fitted
+
+    automl = AutoMLSearch(problem_type='binary')
+    automl.search(X, y)
+    assert automl.best_pipeline._is_fitted
```

Comment:
Sounds good.

Since `_is_fitted` is private, the future-proof way to check this here and in the other tests below would be, if you expect the pipeline to *not* be fitted:
```python
with pytest.raises(SomeException, match="not fitted"):
    automl.best_pipeline.predict(X)
```
or in the case when you expect the pipeline to be fitted:
```python
automl.best_pipeline.predict(X)
```